src/integration-tests/bash/cleanup.sh

```diff
@@ -63,248 +50,74 @@ USER_PROJECTS_DIR="$RESULT_DIR/user-projects"
-          --all-namespaces=true 2>&1 \
-          | egrep -e "($3)" | sort > $resfile_yes 2>&1
-      artcount_yes="`cat $resfile_yes | wc -l`"
-
-      # leftover non-namespaced artifacts
-      kubectl get $2 \
-          -o=jsonpath='{range .items[*]}{.kind}{"/"}{.metadata.name}{"\n"}{end}' \
-          --all-namespaces=true 2>&1 \
-          | egrep -e "($3)" | sort > $resfile_no 2>&1
-      artcount_no="`cat $resfile_no | wc -l`"
-
-      artcount_total=$((artcount_yes + artcount_no))
-
-      mnow=`date +%s`
-
-      if [ $((artcount_total)) -eq 0 ]; then
-        echo "@@ No artifacts found."
-        return 0
-      fi
-
-      if [ "$iteration" = "first" ]; then
-        # in the first iteration we just wait to see if artifacts go away on there own
-
-        echo "@@ Waiting for $artcount_total artifacts to delete.  Wait time $((mnow - mstart)) seconds (max=$maxwaitsecs).  Waiting for:"
-
-        cat $resfile_yes | awk '{ print "n=" $1 " " $2 }' 
-        cat $resfile_no | awk '{ print $1 }'
-
-      else
-        # in the second thirty seconds we try to delete remaining artifacts
+function getResWithLabel {
 
-        echo "@@ Trying to delete ${artcount_total} leftover artifacts, including ${artcount_yes} namespaced artifacts and ${artcount_no} non-namespaced artifacts, wait time $((mnow - mstart)) seconds (max=$maxwaitsecs)."
+  # first, let's get all namespaced types with -l $LABEL_SELECTOR
+  kubectl get $NAMESPACED_TYPES \
+          -l "$LABEL_SELECTOR" \
+          -o=jsonpath='{range .items[*]}{.kind}{" "}{.metadata.name}{" -n "}{.metadata.namespace}{"\n"}{end}' \
+          --all-namespaces=true >> $1
 
-        if [ ${artcount_yes} -gt 0 ]; then 
-          cat "$resfile_yes" | while read line; do
-            local args="`echo \"$line\" | awk '{ print "-n " $1 " delete " $2 " --ignore-not-found" }'`"
-            echo "kubectl $args"
-            kubectl $args
-          done
-        fi
-
-        if [ ${artcount_no} -gt 0 ]; then 
-          cat "$resfile_no" | while read line; do
-            echo "kubectl delete $line --ignore-not-found"
-            kubectl delete $line --ignore-not-found
-          done
-        fi
-
-      fi
-
-      if [ $((mnow - mstart)) -gt $((maxwaitsecs)) ]; then
-        if [ "$iteration" = "first" ]; then
-          echo "@@ Warning:  ${maxwaitsecs} seconds reached.   Will try deleting unexpected resources via kubectl delete."
-        else
-          echo "@@ Error:  ${maxwaitsecs} seconds reached and possibly ${artcount_total} artifacts remaining.  Giving up."
-        fi
-        break
-      fi
-
-      sleep 5
-    done
-  done
-  return 1
-}
-
-function kubectlDeleteF {
-   if [ -f "$1" ]; then
-      kubectl delete -f "$1" --ignore-not-found
-   else
-      echo @@ File \"$1\" not found.  Skipping kubectl delete -f.
-   fi
+  # now, get all non-namespaced types with -l $LABEL_SELECTOR
```

Comment:
I think it's good to replace 'orderly delete' with a label based deletes, but I think the 'genericDelete' step should not be removed and should still be the final step.   IMO, it's still needed to help guarantee a full cleanup.   Cleanup.sh is used by wercker and on an ad-hoc basis after test runs, so it seems important to have it cleanup even if someone forgets to label a new resource properly.

Also, it's important IMO to do generic deletes in a timer loop, since sometimes 'out-of-order' deletes aren't helpful  -- for example,. a resource can be automatically created by another resource that hasn't been deleted yet.   See the original version of cleanup.sh or see the delete-domain script for the pattern.

Reply:
It's equally important to do orderly deletes in a retry loop for similar reasons.   The delete-domain script already does this internally for domains, but the same needs to be done for the operator label based delete...

Reply:
Finally, the delete-domain script tries to wait for a clean shutdown, but this wait isn't needed for 'cleanup.sh' which has no need for one.   To keep the running time short, maybe it'd be good to add a new option to the delete-domain script '-f' to cause it to skip directly to its 'Phase 3'.

Reply:
One more comment (sorry).  It looks like the code for deleting namespaces was removed from cleanup.sh -- this code should be restored as well.   

To summarize:  it's good that cleanup.sh now calls delete-domain, and that it now deletes operator stuff via labels, but (A) it should also retain its old logic for cleaning up anything (the old 'generic delete') in case label deletes fail, (B) the new operator label delete needs to be retried up to some number of seconds  (similar to the logic in delete-domain), and (C) it needs to retain its old logic for deleting namespaces.

Reply:
all your comments are addressed except (A) it should also retain its old logic for cleaning up anything (the old 'generic delete') in case label deletes fail.
If all the resources are labeled properly, deleting with label can always succeed. If some resources don't have the proper label and cause wercker fail, what need to be fixed is the resource.

Now the cleanup.sh doesn't  call the delete-domain script. Seems to me the logic in delete-domain script is more complicated than it should be. 
One thing I'm not sure is the differences between the two approaches below to delete domain:
1. setting domain's startupControl to NONE to let wls operator shutdown the domain
2. delete domain related resources directly
The delete-domain script can be cleanup after understanding this better.